### PR TITLE
added securityContext to manager's pod

### PR DESCRIPTION
**Pull request by idanl21**
added - https://github.com/Opster/opensearch-k8s-operator/issues/457

### CODE REVIEW

File: charts/opensearch-operator/templates/opensearch-operator-controller-manager-deployment.yaml [hunk 1]
```diff
@@ -43,6 +43,8 @@ spec:
         image: "{{ .Values.manager.image.repository }}:{{ .Values.manager.image.tag | default .Chart.AppVersion }}"
         name: operator-controller-manager
         imagePullPolicy: "{{ .Values.manager.image.pullPolicy }}"
+        securityContext:
```
Comment (saketmht) [inline charts/opensearch-operator/templates/opensearch-operator-controller-manager-deployment.yaml:46]: The securityContext field already exists below. I don't think we need to create another one. Just add the value that already exists in the values.yaml

**Comment by saketmht** [comment]
We need a field `podSecurityContext` for the entire deployment's securityContext as well
https://github.com/Opster/opensearch-k8s-operator/blob/4edd47a61f05e2a3b2e38d826f9fb5ca33e55f1f/charts/opensearch-operator/templates/opensearch-operator-controller-manager-deployment.yaml#L66
and securityContext for the `kubeRbacProxy` container would also be nice.

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]
I agree with @saketmht's review.

File: charts/opensearch-operator/values.yaml [hunk 2]
```diff
@@ -37,6 +37,8 @@ manager:
 #    fsGroup: 2000
   # Set this to false to disable the experimental parallel recovery in case you are experiencing problems
   parallelRecoveryEnabled: true
+  allowPrivilegeEscalation: false
```
Comment (saketmht) [inline charts/opensearch-operator/values.yaml:40]: This should be under `securityContext`

**Review by saketmht — CHANGES_REQUESTED** [review]
The changes now are only for the `manager` container. Like I mentioned in my earlier comment, I think we should support setting security context for the entire deployment as well as the `kuberbacproxy` container.

File: charts/opensearch-operator/values.yaml [hunk 3]
```diff
@@ -72,6 +86,10 @@ kubeRbacProxy:
     timeoutSeconds: 3
     initialDelaySeconds: 10
 
+    securityContext:
```
Comment (swoehrl-mw) [inline charts/opensearch-operator/values.yaml:89]: Wrong indentation.

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]

File: charts/opensearch-operator/values.yaml [hunk 4]
```diff
@@ -30,9 +34,19 @@ manager:
...
+
   parallelRecoveryEnabled: true
 
+# Set securityContext for manager
+  securityContext:
+    allowPrivilegeEscalation: false
+#    runAsUser: 1000
+#    runAsGroup: 3000
+#    fsGroup: 2000
+  # Set this to false to disable the experimental parallel recovery in case you are experiencing problems
```
Comment (saketmht) [inline charts/opensearch-operator/values.yaml:46]: This comment should be above parallelRecoveryEnabled flag at Line37

**Comment by saketmht** [comment]
Should the file diff be this large? Did the merge/rebase go wrong? The indentation error as swoehrl-mw mentioned also needs to be fixed.

**Comment by saketmht** [comment]
@idanl21 Any updates on this PR? Would really like this feature to be checked in as we require setting security context for the pods in our environment.

**Comment by saketmht** [comment]
> Hey friends, @saketmht @swoehrl-mw What about that PR?

@idanl21 there is still a mistake in the values.yaml Line 89. The indentation is incorrect for securityContext. And I am still not sure why the file diff is so large compared to main branch.

**Comment by saketmht** [comment]
@idanl21 Are you still working on this? We would really like this feature to be included in the next release. If you don't have bandwidth, do you want me to raise a PR for this?

**Comment by idanl21** [comment]
Hey @saketmht 
Sorry on the delay, but that feature is allready finished, waiting only for review :) 
You are more then welcome to take a look and we will release it in next version

**Comment by swoehrl-mw** [comment]
Hi @idanl21. There is something weird with your PR, you have diffs for things that are already in main (for example `securityContext` for `GeneralConfig` and `DashboardsConfig`). Please check your PR and either fix it or recreate it. As it is now it cannot be reviewed or merged.

**Comment by swoehrl-mw** [comment]
Superseded by PR #561.

**Pull request closed without merging**